src/course-unit/course-xblock/xblock-content/iframe-wrapper/iframe-wrapper.js

```diff
@@ -17,14 +17,31 @@ import {
  *                   Only required for legacy XBlocks that don't declare their
  *                   JS and CSS dependencies properly.
  * @param type The XBlock's type (openassessment, discussion, video, etc.)
+ * @param stylesWithContent Custom CSS global for all XBlock's.
  */
-export default function wrapBlockHtmlForIFrame(html, sourceResources, studioBaseUrl, type) {
+export default function wrapBlockHtmlForIFrame(
+  html,
+  sourceResources,
+  studioBaseUrl,
+  type,
+  stylesWithContent,
+) {
   const resources = normalizeResources(sourceResources);
 
   /* Extract CSS resources. */
   const cssUrls = filterAndExtractResources(resources, 'url', 'text/css');
   const sheets = filterAndExtractResources(resources, 'text', 'text/css');
 
+  let additionalCSSTags = '';
+  additionalCSSTags += stylesWithContent
+    .map(sheet => {
+      if (Array.isArray(sheet)) {
+        return sheet.map(subSheet => `<style>${subSheet}</style>`).join('\n');
+      }
+      return `<style>${sheet}</style>`;
+    })
+    .join('\n');
```

Comment:
```suggestion
  const additionalCSSTags = stylesWithContent.flatMap(sheet => `<style>${sheet}</style>`).join('\n');
```

In JavaScript, strings are immutable. Every time you use `+=` or concatenation, a new string is created in memory, and the old one is potentially discarded.  This can lead to excessive memory usage if you do it often in a loop.

By the way `flatMap` can help with nested arrays to make it flat and map it at the same time.

Reply:
Yeah, good catch! `flatMap` is very good to apply here!